syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -166,33 +182,42 @@ def _split_number(self, number, bits):
             number_parts.append(part * sign)
         return number_parts[::-1]
 
-    def _restore_number(self, number_parts, bits):
-        """Rebuild a number from a numpy array,
+    @staticmethod
+    def _internal_representation_to_large_ints(tensor, precision) -> np.array:
+        """Creates an numpy array containing the objective large numbers."""
+        ndarray = tensor.numpy()
+        ndarray = ndarray.reshape(-1, ndarray.shape[-1])
+        result = []
+        for elem in ndarray:
+            result.append(LargePrecisionTensor._restore_large_number(elem, precision))
+        return np.array(result)
+
+    @staticmethod
+    def _restore_large_number(number_parts, bits) -> int:
+        """Rebuilds a number from a numpy array.
 
         Args:
             number_parts (ndarray): the numpy array of numbers representing the original one.
             bits (int): the bits used in the split.
 
         Returns:
-            Number: the original number.
+            Number: the large number represented by this tensor
         """
         base = 2 ** bits
         n = 0
-        # Using tolist() to allow int type. Terrible performance :(
         for x in number_parts.tolist():
             n = n * base + x
-        return n / (self.base ** self.precision_fractional)
+        return n
 
 
-# The internal precision used to decompose the large numbers is half of the size of the type.
-# This is because we are not considering negative numbers when decomposing.
+# The internal precision used to decompose the large numbers is the size of the type - 1.
 internal_precision = {
```

Comment:
make a @property in the class to allow `self.internal_precision` instead of `internal_precision[self.internal_type]`, it's more pythonic (and will be useful)